### ISSUE TITLE
Code block line highlighting not highlighting correct line

### ISSUE THREAD

**Issue by jacobclevenger** [vuejs/vitepress#63]
**Describe the bug**
When creating a code block using line highlighting, it does not highlight the correct line. Instead it seems to highlight the line between the desired line and the previous line.

**To Reproduce**

Create a new vitepress project. In any markdown page create a codeblock and specify the line to highlight. Check the output and the correct line will not be highlighted.

Example:

```md
## Usage
```html{3}
<template>
    <div>
        <kro-button @click="delete">Delete Cake</kro-button>
    </div>
</template>

<script setup>
    import { useDialog } from '@black-kro/ui';

    const { createDialog } = useDialog();

    export const delete = () => {
        createDialog({
            title: 'Delete Cake?',
            message: 'Are you sure you would like to delete the cake?',
        })
            .then(() => {
                // Delete Cake.
            })
            .catch(() => {
                // Don't delete Cake.
            })
    }
</script>
```

Output:
![image](https://user-images.githubusercontent.com/7064956/89157593-c3526d80-d53a-11ea-83b0-875afd8b04ab.png)

**Expected behavior**
Line 3 should be highlighted.

**System Info**
- vitepress version: 0.5.0
- vite version: 1.0.0-rc.3
- Node version: 14.7.0
- OS version: Windows 10

**Comment by jacobclevenger** [comment]
Potential Fix:

By modifying the `.highlight-lines` class in https://github.com/vuejs/vitepress/blob/master/src/client/theme-default/styles/code.css this can be fixed.

Replace the current `.highlight-lines` class with
```css
.highlight-lines {
  font-size: 0.9em;
  user-select: none;
  padding-top: 2em;
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  line-height: 1.5;
}
```

Replacing the `padding-top: 1.3rem` with `padding-top: 2em` seems to fix the problem for me at least.

**Comment by kiaking** [comment]
Yap, seems like we have issue on code block stylings, when adding line numbers options too. Also, it seems like style changes on `dev` and `build` too 🤔

**Comment by kiaking** [comment]
This is fixed with v0.9.0 🙌

**Comment by Hazlank** [comment]
@kiaking I read vite doc,then i saw the same bug,i clone the project and upgrade to v.0.9.0, even v0.15.0,is's still display error
![1628079064(1)](https://user-images.githubusercontent.com/15724316/128178494-d48b95b9-45a5-481c-9967-1505ab1aaeff.png)
![1628079041(1)](https://user-images.githubusercontent.com/15724316/128178501-bd7c8a5e-e314-4208-bdc4-d5eacb2cc3be.png)
why?
